ThunderCloud/AchievementDisplayView.swift

```diff
@@ -39,83 +185,157 @@ open class AchievementDisplayView: UIView, AchievementDisplayable {
         
-        let availableSize = CGSize(width: frame.size.width - AchievementDisplayView.contentEdgePadding.left - AchievementDisplayView.contentEdgePadding.right, height: CGFloat.greatestFiniteMagnitude)
-        let titleSize = titleLabel.sizeThatFits(availableSize)
-        titleLabel.frame = CGRect(
-            x: AchievementDisplayView.contentEdgePadding.left,
-            y: badgeImageView.frame.minY - titleSize.height - AchievementDisplayView.labelImageSpacing,
-            width: frame.width - AchievementDisplayView.contentEdgePadding.left - AchievementDisplayView.contentEdgePadding.right,
-            height: titleSize.height
-        )
+        addSubview(stackView)
+    }
+      
+    /// Constrain views
+    private func constrain() {
+        stackView.translatesAutoresizingMaskIntoConstraints = false
+        progressView.translatesAutoresizingMaskIntoConstraints = false
+        badgeImageView.translatesAutoresizingMaskIntoConstraints = false
+        spaceView.translatesAutoresizingMaskIntoConstraints = false
         
-        let size = subtitleLabel.sizeThatFits(availableSize)
-        subtitleLabel.frame = CGRect(
-            x: AchievementDisplayView.contentEdgePadding.left,
-            y: badgeImageView.frame.maxY + AchievementDisplayView.labelImageSpacing,
-            width: frame.size.width - AchievementDisplayView.contentEdgePadding.left - AchievementDisplayView.contentEdgePadding.right,
-            height: size.height
-        )
+        NSLayoutConstraint.activate([
+            // StackView
+            stackView.centerXAnchor.constraint(equalTo: centerXAnchor),
+            stackView.widthAnchor.constraint(equalTo: widthAnchor, multiplier: Constants.stackViewWidthScale),
+            
+            // BadgeView
+            progressView.widthAnchor.constraint(equalTo: stackView.widthAnchor,
+                                                multiplier: Constants.progressViewWidthScale),
+            progressView.heightAnchor.constraint(equalTo: progressView.widthAnchor),
+            
+            // ImageView
+            badgeImageView.centerXAnchor.constraint(equalTo: progressView.centerXAnchor),
+            badgeImageView.centerYAnchor.constraint(equalTo: progressView.centerYAnchor),
+            badgeImageView.widthAnchor.constraint(equalTo: progressView.widthAnchor,
+                                             multiplier: Constants.imageViewWidthScale),
+            badgeImageView.heightAnchor.constraint(equalTo: badgeImageView.widthAnchor),
+            
+            // SpaceView
+            spaceView.widthAnchor.constraint(equalTo: stackView.widthAnchor),
+            spaceView.heightAnchor.constraint(equalToConstant: 5),
+        ])
         
-        centerSubviewsVertically()
+        if AchievementDisplayView.interfaceOptions.centerVertically {
+            NSLayoutConstraint.activate([
+                stackView.centerYAnchor.constraint(equalTo: centerYAnchor),
```

Comment:
@simonmitchell  Shouldn't need both 
- `top` and `bottom` constraint
- `centerY` constraint